ui/src/lib/asyncHooks.ts

```diff
@@ -0,0 +1,77 @@
+import { AsyncThunkAction, unwrapResult } from '@reduxjs/toolkit';
+import { useCallback, useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { useFlash } from 'src/duck/flashMessages';
+
+interface Options {
+  rethrowError?: boolean;
+  errorTitle?: string;
+  successTitle?: string;
+}
+
+/**
+ * useAsyncThunk return a callback
+ * it return a promise which is resolved/rejected to notify user when the operation is completed via flash message
+ */
```

Comment:
Docs are wrong but I am letting it slide. Let's fix them incrementally as we use these hooks more.
Writing good documentation is an important exercise.